go-controller/pkg/factory/factory.go

```diff
@@ -657,6 +692,18 @@ func NewClusterManagerWatchFactory(ovnClientset *util.OVNClusterManagerClientset
 		wf.efFactory.K8s().V1().EgressFirewalls().Informer()
 	}
 
+	if config.OVNKubernetesFeature.EnableEgressFirewall && config.OVNKubernetesFeature.EnableDNSNameResolver {
```

Comment:
should we move this inside the previous `if config.OVNKubernetesFeature.EnableEgressFirewall {` block?

Reply:
Made the changes.